src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/directionsquestionnaire/HearingContentProvider.java

```diff
@@ -0,0 +1,88 @@
+import uk.gov.hmcts.cmc.claimstore.services.staff.models.ExpertReportContent;
+import uk.gov.hmcts.cmc.claimstore.services.staff.models.HearingContent;
+import uk.gov.hmcts.cmc.claimstore.utils.DateUtils;
+import uk.gov.hmcts.cmc.claimstore.utils.Formatting;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.DirectionsQuestionnaire;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.ExpertReport;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.ExpertRequest;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.RequireSupport;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.UnavailableDate;
+import uk.gov.hmcts.cmc.domain.models.response.YesNoOption;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.toList;
+import static uk.gov.hmcts.cmc.ccd.util.StreamUtil.asStream;
+
+@Component
+public class HearingContentProvider {
+
+    private static final String DISABLED_ACCESS = "Disabled Access";
+    private static final String YES = "Yes";
+    private static final String NO = "No";
+
+    private Function<ExpertReport, ExpertReportContent> mapExpertReport = report ->
+        ExpertReportContent.builder().expertName(report.getExpertName())
+            .expertReportDate(Formatting.formatDate(report.getExpertReportDate()))
+            .build();
+    private Function<UnavailableDate, String> mapToISOFullStyle = unavailableDate ->
+        Optional.ofNullable(unavailableDate)
+            .map(UnavailableDate::getUnavailableDate)
+            .map(DateUtils::toISOFullStyle).orElse("");
+
+    private BiConsumer<RequireSupport, HearingContent.HearingContentBuilder> mapSupportRequirement =
+        (support, builder) -> {
+            List<String> supportNeeded = new ArrayList<>();
+            support.getLanguageInterpreter().ifPresent(supportNeeded::add);
+            support.getSignLanguageInterpreter().ifPresent(supportNeeded::add);
+            support.getDisabledAccess()
+                .map(YesNoOption::name)
+                .filter(access -> access.equals(YesNoOption.YES.name()))
+                .ifPresent(x -> supportNeeded.add(DISABLED_ACCESS));
+            support.getOtherSupport().ifPresent(supportNeeded::add);
+            builder.supportRequired(supportNeeded);
+        };
+
+    private BiConsumer<ExpertRequest, HearingContent.HearingContentBuilder> mapExpertRequest =
+        (expertRequest, builder) -> {
+            builder.hasExpertReport(YES);
+            builder.courtPermissionForExpertReport(YES);
+            builder.reasonWhyExpertAdvice(expertRequest.getReasonForExpertAdvice());
+            builder.expertExamineNeeded(expertRequest.getExpertEvidenceToExamine());
+        };
+
+    public Function<DirectionsQuestionnaire, HearingContent> mapDirectionQuestionnaire =
```

Comment:
Implementing functional interface here and not used as the assignment target for a lambda expression or method reference, appears misuse of functional interface. This can be plane vanilla java  method. i.e. 
```
HearingContent  mapDirectionQuestionnaire(DirectionsQuestionnaire questionnaire) {
...
}
```

Reply:
Yeah. I am changing these to java methods.